src/_http/errors.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+var util = require('../_util')
+
+/**
+ * Thrown by HttpClient when request hits specified timeout.
+ *
+ * @param {?string} message
+ * @extends Error
+ * @constructor
+ */
+function TimeoutError(message) {
+  Error.call(this)
+
+  this.message = message || 'Request aborted due to timeout'
+  this.isTimeoutError = true
+}
+
+util.inherits(TimeoutError, Error)
+
+/**
+ * Thrown by HttpClient when request is aborted via Signal interface.
+ *
+ * @param {?string} message
+ * @extends Error
+ * @constructor
+ */
+function AbortError(message) {
+  Error.call(this)
+
+  this.message = message || 'Request aborted'
+  this.isAbortError = true
+}
+
+util.inherits(AbortError, Error)
+
+module.exports = {
+  TimeoutError: TimeoutError,
+  AbortError: AbortError,
+}
```

Comment:
Q: Are these new or were they existing errors?

Reply:
These are new errors shared across both adapters, to expose the same error API from `HttpClient` outside. E.g. [here](https://github.com/fauna/faunadb-js/pull/396/files#diff-a36fdfce533ef260dcd9afc30b77d10eecb0c6e2711fbcbef70e66a1ca5ab7d5R133) we assert `AbortError` no matter what underlying adapter is used as they share the same errors for abort / timeout

Reply:
These are new errors shared across both adapters, to expose the same error API from `HttpClient` outside. E.g. [here](https://github.com/fauna/faunadb-js/pull/396/files#diff-a36fdfce533ef260dcd9afc30b77d10eecb0c6e2711fbcbef70e66a1ca5ab7d5R133) we assert `AbortError` no matter what underlying adapter is used as they share the same errors for abort / timeout